lib/kitchen/directions/bake_injected_exercise/bake_injected_exercise_question.rb

```diff
@@ -41,11 +42,15 @@ def bake(question:, number:, only_number_solution:)
         end
       end
 
+      context = question.ancestor(:injected_exercise).exercise_context&.cut&.paste if figure_reference
```

Comment:
@kswanson33 I have tried refactored the code with your requested changes but I've got stuck in this particular place and got such error: 
![image](https://user-images.githubusercontent.com/55203598/139940007-7e01c77f-abe0-41b5-9ae0-6c5886fc1924.png)
In my recipe I'm using this loop, where it is not pointing to `injected_exercises but `exercise_question``: 
![image](https://user-images.githubusercontent.com/55203598/139940751-c14507a8-6d7e-4971-b3c7-8e0e5d629105.png)
and I guess this is the reason why I can't select an ancestor for the `exercise_question` of given type (`:injected_exercise`). Do you think there is a need to change the `BakeInjectedExerciseQuestion`  to point `injected_exercises` instead of  `exercise_question`? Or maybe you can see a different approach here, rewrite the loop in the recipe somehow? Let me know if you will have some suggestions.

Reply:
@kswanson33 could you also explain what did you mean by `single-part exercises`?

Reply:
Small EDIT:
I have also tried to use the element class `InjectedExerciseElement` itsef: 
![image](https://user-images.githubusercontent.com/55203598/140101603-c725057a-4248-42ea-bd7f-84e67521af99.png)
but also got error: 
![image](https://user-images.githubusercontent.com/55203598/140101753-61b7a3cc-824c-453f-9607-3b33d42e44f9.png)